apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/dao/DexOfferTable.java

```diff
@@ -0,0 +1,81 @@
+package com.apollocurrency.aplwallet.apl.exchange.dao;
+
+import com.apollocurrency.aplwallet.apl.core.app.Blockchain;
+import com.apollocurrency.aplwallet.apl.core.db.DbKey;
+import com.apollocurrency.aplwallet.apl.core.db.EntityDbTable;
+import com.apollocurrency.aplwallet.apl.core.db.LongKey;
+import com.apollocurrency.aplwallet.apl.core.db.LongKeyFactory;
+import com.apollocurrency.aplwallet.apl.core.db.dao.mapper.DexOfferMapper;
+import com.apollocurrency.aplwallet.apl.exchange.model.DexOffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Objects;
+
+/**
+ * Implemented for backward compatibility with rollback function in the DerivedDbTable.
+ * Use DexOfferDao
+ */
+@Deprecated
+@Singleton
+public class DexOfferTable  extends EntityDbTable<DexOffer> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DexOfferTable.class);
+
+    static final LongKeyFactory<DexOffer> KEY_FACTORY = new LongKeyFactory<>("transaction_id") {
+        @Override
+        public DbKey newKey(DexOffer offer) {
+            return new LongKey(offer.getTransactionId());
+        }
+
+        @Override
+        public DexOffer newEntity(DbKey dbKey) {
+            DexOffer offer = new DexOffer();
+            offer.setTransactionId(((LongKey) dbKey).getId());
+            return offer;
+        }
+    };
+
+
+    private static final String TABLE_NAME = "dex_offer";
+    private final Blockchain blockchain;
+    private DexOfferMapper dexOfferMapper;
```

Comment:
mapper could be static final

Reply:
RowMapper.map() is not static, so we couldn't do that.